components/nocodb/nocodb.app.mjs

```diff
@@ -1,11 +1,146 @@
+import { Api } from "nocodb-sdk";
+
 export default {
   type: "app",
   app: "nocodb",
-  propDefinitions: {},
+  propDefinitions: {
+    projectId: {
+      type: "string",
+      label: "Project Id",
+      description: "The id of the project",
+      async options() {
+        const { list } = await this.listProjects();
+        return list.map((project) => ({
+          label: project.title,
+          value: project.id,
+        }));
+      },
+    },
+    tableName: {
+      type: "string",
+      label: "Table Name",
+      description: "The name of the table",
+      withLabel: true,
+      async options({ projectId }) {
+        const { list } = await this.listTables({
+          projectId,
+        });
+        return list.map((table) => ({
+          label: table.title,
+          value: table.id,
+        }));
+      },
```

Comment:
You can use the `page` or `prevContext` prop to handle pagination for `tableName` prop here.